### PR TITLE
BaSP-TG-72: Timesheet list

### PR DESCRIPTION
**Pull request by GuidoCerioni**

### CODE REVIEW

File: src/Components/TimeSheets/index.js [hunk 1]
```diff
@@ -1,43 +1,108 @@
...
-  const deleteTimeSheet = async (_id) => {
-    await fetch(`${process.env.REACT_APP_API_URL}/time-sheets`, { method: 'DELETE' });
-    setList([...list.filter((timeSheet) => timeSheet._id !== _id)]);
-  };
+  function handleDeleteTimesheet(id) {
+    setDeleteId(id);
+    setIsOpen(true);
+  }
+
+  async function deleteTimesheet(id) {
```
Comment (AlejandroSantini) [inline src/Components/TimeSheets/index.js:56]: You should add setLoading (true), then setLoading (false)

**Review by AlejandroSantini — APPROVED** [review]
Nice job!

File: src/Components/TimeSheets/index.js [hunk 2]
```diff
@@ -1,43 +1,108 @@
...
-      <List list={list} setList={setList} deleteTimeSheet={deleteTimeSheet} />
-      <a className={styles.btn} href="/time-sheets/form">
-        Add New Time Sheet
-      </a>
+      {loading ? (
+        <LoadingScreen />
+      ) : (
+        <>
+          <Modal
+            modalTitle="Are you sure you want to delete?"
```
Comment (AlejandroSantini) [inline src/Components/TimeSheets/index.js:80]: The same as commented in the other PR

File: src/Components/TimeSheets/time-sheets.module.css [hunk 3]
```diff
@@ -1,26 +1,13 @@
 .container {
```
Comment (AlejandroSantini) [inline src/Components/TimeSheets/time-sheets.module.css:1]: The same as commented in the other PR

**Review by AlejandroSantini — CHANGES_REQUESTED** [review]
Correct comments

**Review by AlejandroSantini — APPROVED** [review]
Exelent job!

File: src/Components/TimeSheets/index.js [hunk 4]
```diff
@@ -55,36 +53,48 @@ const TimeSheets = () => {
...
+      setBothDatas(data.filter((item) => item._id !== id));
       setIsOpen(false);
     }
-  }
-
-  function search(value) {
-    setSearchQuery(value);
-    const result = untouchedData.filter((employee) => employee._id.includes(value));
-    setTimesheets(result);
+    setLoading(false);
```
[the remark this answers is not in the document]
Comment (GuidoCerioni) [reply src/Components/TimeSheets/index.js:77]: yes, it is there, but it doesnt shows correctly

File: src/Components/TimeSheets/index.js [hunk 5]
```diff
@@ -1,43 +1,116 @@
...
+    const result = untouchedData.filter((employee) => employee._id.includes(value));
+    setData(result);
+  }
+
+  async function deleteTimesheet(id) {
+    setLoading(true);
+    const response = await fetch(`${URL}/${id}`, { method: 'DELETE' });
+    const responseJson = await response.json();
+    if (responseJson.error) {
+      alert('error');
```
Comment (HernanCazeneuve) [inline src/Components/TimeSheets/index.js:72]: use modal instead of alert

File: src/Components/TimeSheets/index.js [hunk 6]
```diff
@@ -1,43 +1,116 @@
...
+        setLoading(true);
+        const response = await fetch(`${URL}`);
+        const { message, data, error } = await response.json();
+        if (!error) {
+          setBothDatas(data);
+        } else {
+          throw new Error(message);
+        }
+      } catch (error) {
+        console.log('error', error);
```
Comment (HernanCazeneuve) [inline src/Components/TimeSheets/index.js:43]: try not to use console log

File: src/Components/TimeSheets/index.js [hunk 7]
```diff
@@ -1,43 +1,116 @@
...
-  if (loading) {
-    return <p>Data is loading...</p>;
+  function setBothDatas(data) {
+    setData(data);
+    setUntouchedData(data);
+  }
+
+  function search(value) {
+    setSearchQuery(value);
+    const result = untouchedData.filter((employee) => employee._id.includes(value));
```
Comment (CarVigil) [inline src/Components/TimeSheets/index.js:63]: I think it's better to search by first or last name instead of id. What do you think?

**Review by CarVigil — CHANGES_REQUESTED** [review]

**Review by AlejandroSantini — DISMISSED** [review]
Good job!

**Review by CarVigil — APPROVED** [review]

**Review by FiolAxel — APPROVED** [review]

**Review by HiginiaMed — APPROVED** [review]

**Review by AlejandroSantini — APPROVED** [review]
Nice!

**Review by Gearyandres94 — APPROVED** [review]

**Pull request merged**